code/game/turfs/open/floor.dm

```diff
@@ -207,23 +207,23 @@
 
 /turf/open/floor/singularity_pull(S, current_size)
 	..()
+	var/do_et = FALSE
 	if(current_size == STAGE_THREE)
 		if(prob(30))
-			if(floor_tile)
-				new floor_tile(src)
-				make_plating(TRUE)
+			do_et = TRUE
 	else if(current_size == STAGE_FOUR)
```

Comment:
if you're refactoring this you might also look into converting this into a switch statement for clarity

(also pls use descriptive var names if possible, after some time it just becomes bothersome to interpret)

Reply:
Gottcha